### PR TITLE
add command to preview deployments

### LINKED ISSUE #432: Preview the resources to the user that would be provisioned
Similar to previewing prior to deleting resources, can we also preview what will be deployed.

Users should be able to abort the operation if they wish.

### PR DESCRIPTION
**Pull request by HomelessDinosaur**
Closes: https://github.com/nitrictech/cli/issues/432

Output Example (changing the name of the API):
Screen Shot 2022-12-19 at 1 51 15 pm

### CODE REVIEW

**Comment by HomelessDinosaur** [comment]
Considering removing the "View Live" line as well, as that is pulumi specific and will require them to sign in.

**Comment by tjholm** [comment]
> Considering removing the "View Live" line as well, as that is pulumi specific and will require them to sign in.

@HomelessDinosaur this likely only appears if they're signed into pulumi. It would be interesting the see the result for a local login.

**Comment by HomelessDinosaur** [comment]
> @HomelessDinosaur this likely only appears if they're signed into pulumi. It would be interesting the see the result for a local login.

You're right, tested locally whilst logged out and the link wasn't there.

**Review by tjholm — APPROVED** [review]

File: pkg/provider/pulumi/generator.go [hunk 1]
```diff
@@ -288,3 +288,48 @@ func (a *pulumiDeployment) Down(log output.Progress) (*types.Summary, error) {
...
+	for _, line := range lines {
+		if !strings.Contains(line, "pulumi:") { // Remove pulumi specific resources
+			color := colorWhite
+			if strings.Contains(line, " +") { // Creating a resource
+				color = colorGreen
+			} else if strings.Contains(line, " -") { // Deleting a resource
+				color = colorRed
+			} else if strings.Contains(line, " ~") { // Updating a resource
+				color = colorYellow
+			}
```
Comment (tjholm) [inline pkg/provider/pulumi/generator.go:324]: There may be cases where some lines could contain this substring (I think the leading space will probably occlude most though)

**Pull request merged**